crates/evm/src/evm/executor.rs

```diff
@@ -87,7 +115,8 @@ pub(crate) fn execute_multiple_tx<
                 hex::encode(tx.hash())
             )))
         } else {
-            evm.transact_commit(tx)
+            evm.commit(result_and_state.state);
+            Ok(result_and_state.result)
         };
         cumulative_gas_used += result.as_ref().map(|r| r.gas_used()).unwrap_or(0);
```

Comment:
move this into the else block

Reply:
why?

Reply:
easier to read bc instead of as_ref map unwrap or 

you have 

cumulative_gas_used += result.gas_used() :D